include/mlir/SPIRV/SPIRVTypes.h

```diff
@@ -52,9 +53,28 @@ enum Kind {
 };
 }
 
+// Provides a common base class between SPIR-V ArrayType, SPIR-V StructType.
+class CompositeType : public Type {
+public:
+  using ImplType = detail::CompositeTypeStorage;
```

Comment:
I think we can use `DefaultTypeStorage` here to avoid defining `CompositeTypeStorage` given that we are not storing anything inside this base class actually?

Reply:
Fixed.